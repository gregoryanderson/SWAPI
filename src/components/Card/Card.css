.card__article {
    background-image: url('https://png.pngtree.com/thumb_back/fh260/background/20190510/pngtree-starry-sky-white-blue-stars-image_111634.jpg');
    background-size: cover;
    box-sizing: border-box;
    border: 3px solid black;
    padding: 5px;
    margin: 5px;
}

header {
    display: flex;
    justify-content: space-between;
}