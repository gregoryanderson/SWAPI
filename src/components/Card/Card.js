import React from 'react';
import './Card.css';


const Card = ({ data, addFavorite, type }) => {
    console.log('pear', type)
    const displayCards = data.map(card => {
        const {personLanguage, name, personPlanet, personPlanetPopulation, personSpecies, vehicleClass, model, passengers, climate, population, terrain, residents} = card;
        // if (residents) {const displayResidents = residents.map(resident => {
        //   return <p className="resident">{resident || null}</p>
        // })}
        return (
        <article className="card__article" id={name}>
            <header className="card__header">
                <h3 className="card__h3">{name || null}</h3>
                <img src='http://images2.wikia.nocookie.net/__cb20080228205028/starwars/images/thumb/7/71/Redstarbird.svg/1600px-Redstarbird.svg.png' height='40px' width='40px' alt='Star Wars Favorite Button' onClick={() => addFavorite({name}, type)}/>
            </header>
            <div className="card__div">
                <p className="card__p">{personPlanet || null}</p>
                <p className="card__p">{personSpecies || null}</p>
                <p className="card__p">{personPlanetPopulation || null}</p>
                <p className="card__p">{personLanguage || null}</p>
                <p className="card__p">{model || null }</p>
                <p className="card__p">{vehicleClass || null}</p>
                <p className="card__p">{passengers || null}</p>
                <p className="card__p">{climate || null}</p>
                <p className="card__p">{population || null}</p>
                <p className="card__p">{terrain || null}</p>
                <p className="card__p">{residents || null}</p>
            </div>
        </article>
    )
  })
  return (
    <>
        {displayCards}
    </>
  )
}


// function Card (props) {
//     console.log('card', props)
//     return (
//         <article className="card__article">
//             <header className="card__header">
//                 <h3>{props.name}</h3>
//                 <img src='http://images2.wikia.nocookie.net/__cb20080228205028/starwars/images/thumb/7/71/Redstarbird.svg/1600px-Redstarbird.svg.png' height='40px' width='40px' alt='Star Wars Favorite Button'/>
//             </header>
//             <div>
//                 <p>{props.homeworld || null}</p>
//                 <p>{props.species || null}</p>
//                 <p>{props.population || null}</p>
//             </div>
//         </article>
//     )
// }


export default Card;

//think about fragments