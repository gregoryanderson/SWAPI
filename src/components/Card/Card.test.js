import React from "react";
import { shallow } from "enzyme";
import Card from './Card'


describe("Card", () => {
    it("should match the snapshot with all the data passed through", () => {
      const fakePeopleData = [{
        isFavorite: false,
        name: "Luke Skywalker",
        personLanguage: "Galatic Basic", 
        personPlanet: "Tatooine",
        personPlanetPopulation: "200000",
        personSpecies: "Human",
        type: "people"
      },
      {
        isFavorite: true,
        name: "R2-D2",
        personLanguage: "n/a", 
        personPlanet: "Naboo",
        personPlanetPopulation: "4500000",
        personSpecies: "Droid",
        type: "people"
      }]
      const wrapper = shallow(<Card 
      data={fakePeopleData}
      addFavorite={jest.fn()}
      type={"people"}
      />);
      expect(wrapper).toMatchSnapshot();
    });
})