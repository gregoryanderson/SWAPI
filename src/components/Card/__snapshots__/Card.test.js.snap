// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all the data passed through 1`] = `
<Fragment>
  <article
    className="card__article"
    id="Luke Skywalker"
    type="people"
  >
    <header
      className="card__header"
    >
      <h3
        className="card__h3"
      >
        Luke Skywalker
      </h3>
      <img
        alt="Star Wars Favorite Button"
        height="40px"
        onClick={[Function]}
        src="http://images2.wikia.nocookie.net/__cb20080228205028/starwars/images/thumb/7/71/Redstarbird.svg/1600px-Redstarbird.svg.png"
        width="40px"
      />
    </header>
    <div
      className="card__div"
    >
      <p
        className="card__p"
      >
        Tatooine
      </p>
      <p
        className="card__p"
      >
        Human
      </p>
      <p
        className="card__p"
      >
        200000
      </p>
      <p
        className="card__p"
      >
        Galatic Basic
      </p>
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
    </div>
    <div
      className="card__div--button"
    >
      <Link
        className="card__link"
        to="/people/Luke Skywalker"
      >
        <button
          className="card__button"
        >
          Click for More!
        </button>
      </Link>
    </div>
  </article>
  <article
    className="card__article"
    id="R2-D2"
    type="people"
  >
    <header
      className="card__header"
    >
      <h3
        className="card__h3"
      >
        R2-D2
      </h3>
      <img
        alt="Star Wars Favorite Button"
        height="40px"
        onClick={[Function]}
        src="https://vignette.wikia.nocookie.net/starwars/images/0/0e/Whitestarbird.svg/revision/latest?cb=20080310230901"
        width="40px"
      />
    </header>
    <div
      className="card__div"
    >
      <p
        className="card__p"
      >
        Naboo
      </p>
      <p
        className="card__p"
      >
        Droid
      </p>
      <p
        className="card__p"
      >
        4500000
      </p>
      <p
        className="card__p"
      >
        n/a
      </p>
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
      <p
        className="card__p"
      />
    </div>
    <div
      className="card__div--button"
    >
      <Link
        className="card__link"
        to="/people/R2-D2"
      >
        <button
          className="card__button"
        >
          Click for More!
        </button>
      </Link>
    </div>
  </article>
</Fragment>
`;
